packages/api/tests/models/test-rules-model.js

```diff
@@ -358,6 +362,23 @@ test.serial('Creating a kinesis rule where an event source mapping already exist
   }
 });
 
+test.serial('creating an invalid kinesis type rule does not add event mappings', async (t) => {
```

Comment:
Do we care about having the same test to ensure SNS triggers aren't created for invalid rules? I could go either way

Reply:
My thinking was if we tested one that was good, but what do you think? I mean I could go either way too, so took the easy way out.

Reply:
I guess I incline towards being thorough but it does seem repetitive. Alternatively, if all the code in the rules model from lines 237 - 264 was in a separate (private) method, then we could just test using sinon that the method is never called when you try to create an invalid rule.

More importantly, we probably also want to explicitly check that the rule is valid in `rule.update` before updating rule triggers?

Reply:
Yep, like to all of that. Took your suggestions.